src/SFA.DAS.Campaign.UnitTests/Web/Controllers/RegisterInterest/WhenUserRegistersInterest.cs

```diff
@@ -332,6 +334,19 @@ public async Task Then_If_The_Model_Is_Not_Valid_Then_The_User_Data_Collection_S
             //Assert
             _userDataCollection.Verify(x => x.StoreUserData(It.IsAny<UserData>()), Times.Never);
         }
+        
+        [Test]
+        public async Task Then_If_The_Model_Is_Not_Valid_Then_The_Menu_Is_Added()
+        {
+            //Arrange
+            _controller.ModelState.AddModelError("FirstName","First name");
+            
+            //Act
+            await _controller.Index(_registerInterestModel);
+
+            //Assert
+            _mediator.Verify(x=>x.Send(It.IsAny<GetMenuQuery>(), CancellationToken.None), Times.Once);
```

Comment:
As previous comment, is it sufficient to verify the query was sent, or should the check be on the Menu property of the VM?

Reply:
Have added assertion to check view model for menu not being null